packages/sdk/explorer/src/block-explorer.ts

```diff
@@ -362,6 +362,38 @@ export class BlockExplorer {
       }
     }
   }
+  /**
+   * Returns the ContractMapping for the contract at that address, or if a proxy its implementation
+   * by looking up the contract address in CeloScan.
+   * @param address
+   * @returns The ContractMapping for the contract at that address, or undefined
+   */
+  getContractMappingFromCeloScan = async (
+    address: string
+  ): Promise<ContractMapping | undefined> => {
+    const cached = this.addressMapping.get(address)
+    if (cached) {
+      return cached
+    }
+    let metadata = await queryCeloScan(
+      this.kit.connection,
+      this.kit.web3.utils.toChecksumAddress(address)
+    )
+    if (metadata?.implementationAddress) {
+      metadata = await queryCeloScan(
+        this.kit.connection,
+        this.kit.web3.utils.toChecksumAddress(metadata?.implementationAddress)
+      )
```

Comment:
Query Implementation like this or like ? https://github.com/celo-org/developer-tooling/pull/169/files#diff-843fae4063825bb8d1dde9b84e9d6f76dc209c1bde3e6244009ac189efb3bee7R197